### ISSUE TITLE
Trigger Hitbox For "Show Hitboxes"

### ISSUE THREAD

**Issue by BrownieGamingX5** [absoIute/Mega-Hack-Pro-Future#464]
Maybe there could be a feature of where you can see when a trigger's activation point by using the "Show Hitboxes" feature. This also can be from both, the "Touch Triggered", and "Spawn Triggered" options.

**Comment by Nicko2102** [comment]
Oh man this is such a good idea! imo it should have its own separate colour, rather than just taking the "special" hitbox colour, so that it doesn't look like an invisible orb. not sure how the hitbox for spawn triggers would work though...

**Comment by MainConMan** [comment]
Spawn triggered triggers don't have a hitbox I think. You would just see the hitbox of the spawn trigger that spawns the spawn triggered triggers. 

(this message is so hard to understand but idk how to word it better 🙁)

**Comment by MainConMan** [comment]
Spawn and touch triggers would just look like this though. 
![image](https://user-images.githubusercontent.com/95152593/152267679-ed66d162-db2c-48a0-b29b-b319747a8348.png)

**Comment by BrownieGamingX5** [comment]
This is what I am talking about.

**Comment by Nicko2102** [comment]
> 

nice! i never really thought about how non-touch triggers worked, but i like the l o n g hitbox lol

**Comment by Brittank88** [comment]
- Spawn triggers should have no hitbox
- Hitbox outline for touch-triggered triggers already exists but should be made more apparent (when aligned with the grid it's very hard to see)
- Above also applies to the vertical line drawn intersecting the trigger.
- You could argue this suits BetterEdit more, however it piggybacks off of the existing show hitboxes system so I'm not actually sure.

**Comment by Brittank88** [comment]
> Spawn and touch triggers would just look like this though.

The vertical bar having any width is actually inaccurate here as it really is coming into contact with the exact threshold X-pos of the trigger's vertical line that triggers it.

**Comment by Nicko2102** [comment]
> > Spawn and touch triggers would just look like this though.
> 
> The vertical bar having any width is actually inaccurate here as it really is coming into contact with the exact threshold X-pos of the trigger's vertical line that triggers it.

I agree, but maybe this would work better as a visualization. 

> Spawn triggers should have no hitbox.

Yeah i was confused about that, i think the author was thinking about non-touch triggers

**Comment by absoIute** [comment]
Is probably something I'll do when I fix other errors with the hitbox viewer

**Comment by MainConMan** [comment]
I just realized this, but for some reason, some triggers have hitboxes. They only show up in the editor though.
![triggers with hitboxes](https://user-images.githubusercontent.com/95152593/153525896-552f7a4e-cf0c-4931-93ea-58f0671c56de.png)

**Comment by absoIute** [comment]
Yeah, ObjectID 34 & 55 act like this since they weren't implemented properly (I believe they just weren't given their appropriate type-ID so are just treated like blocks, the default)

**Comment by BlockOG** [comment]
For spawn-triggered triggers there could be a line going from the spawn trigger to the spawn-triggered trigger
